tfjs-backend-webgpu/src/reduce_util.ts

```diff
@@ -0,0 +1,134 @@
+/**
+ * @license
+ * Copyright 2019 Google Inc. All Rights Reserved.
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * =============================================================================
+ */
+import {util} from '@tensorflow/tfjs-core';
+
+export interface DataTypeMap {
```

Comment:
Please import DataTypeMap, DataType, UpcastInt32AndMap, UpcastBoolAndMap, UpcastFloat32AndMap, UpcastComplex64AndMap, upcastTypeMap, upcastType, and sumOutType from tfjs-core/src/types (you'll have to import many of these from dist/ because they are not yet exposed as part of the public API).

upcastType should already be available under backend_util, please export the remaining utilities in backend_util, and add a TODO to import from backend_util rather than dist once a new version of core is released.